pinot-common/src/main/java/org/apache/pinot/common/utils/DataTable.java

```diff
@@ -35,9 +35,16 @@
 
   void addException(ProcessingException processingException);
 
+  void addException(int exceptionCode, String exceptionMsg);
+
   Map<Integer, String> getExceptions();
 
-  byte[] toBytes()
+  default byte[] toBytes()
+      throws IOException {
+    return toBytes(false);
+  }
+
+  byte[] toBytes(boolean isStrippingMetadata)
```

Comment:
I don't see the usage of `toBytes(true)`

Reply:
> Suggest adding 2 methods `DataTable toMetadataOnlyDataTable()` and `DataTable toDataOnlyDataTable()` to the `DataTable` interface

what would be the appropriate way to make a toMetadataOnlyDataTable and toDataOnlyDataTable work behind the scene? (if not replicating the object?)

Reply:
I'm thinking of replicating the `DataTable` object but not the content inside. For the `MetadataOnlyDataTable`, instantiate a new data table with just the `_metadata`; for the `DataOnlyDataTable`, instantiate a new data table with everything except for the `_metadata`

Reply:
added the stripMetadata flag to streaming response utils